src/Squirrel/Utility.cs

```diff
@@ -315,12 +314,13 @@ public static IDisposable WithTempFile(out string path, string localAppDirectory
                 path = Path.Combine(di.FullName, name);
 
                 if (!File.Exists(path) && !Directory.Exists(path)) {
+                    File.Create(path).Close();
                     break;
                 }
             }
 
             var thePath = path;
-            return Disposable.Create(() => File.Delete(thePath));
+            return Disposable.Create(() => DeleteFileHarder(thePath, true));
```

Comment:
A bit weird naming for the function. Is it ours or the original one?

Reply:
I have already asked the same question. It’s a Squirrel function used also on other places. The name is funny, but not the most surprising thing about Squirrel…

what it actually does is trying to delete a file twice with some interval in between (in case it was being locked by something else)